docs: update instructions for scaling down store-gateways (#3698)
grafana/mimir: docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md

To guarantee no downtime when scaling down [store-gateways]({{< relref "../architecture/components/store-gateway.md" >}}), complete the following steps:

1. Ensure at least `-store-gateway.sharding-ring.replication-factor` store-gateway instances are running (three when running Grafana Mimir with the default configuration).

> 1. Scale down no more than two store-gateways at the same time. If you enabled [zone-aware replication]({{< relref "../configure/configure-zone-aware-replication.md" >}})

Reviewer: ```suggestion
1. If you enabled [zone-aware replication]({{< relref "../configure/configure-zone-aware-replication.md" >}}), scale down no more than two store-gateways at the same time.
```
Rationale: save the reader a tad of time if they do no have zone-aware replication enabled. (They can stop reading and go on to the next step.)

> 1. Scale down no more than two store-gateways at the same time. If you enabled [zone-aware replication]({{< relref "../configure/configure-zone-aware-replication.md" >}})
>    for store-gateways, you can in parallel scale down any number of store-gateway instances in one zone at a time. Zone-aware replication is enabled by default in the `mimir-distributed` Helm chart.

Reviewer: ```suggestion
1. Scale down no more than two store-gateways at the same time.
  If you enabled [zone-aware replication]({{< relref "../configure/configure-zone-aware-replication.md" >}})
   for store-gateways, you can in parallel scale down any number of store-gateway instances in one zone at a time.
   Zone-aware replication is enabled by default in the `mimir-distributed` Helm chart.
```
Rationale: Separate instruction from additional information or context.

1. Stop the store-gateway instances you want to scale down.

> 1. Remove the stopped instances from the store-gateway ring. In your browser go to the `GET /store-gateway/ring` page that store-gateways expose on their HTTP port

Reviewer 2: This depends on this setting:

```
# Unregister from the ring upon clean shutdown.
  # CLI flag: -store-gateway.sharding-ring.unregister-on-shutdown
  [unregister_on_shutdown: <boolean> | default = true]
```

With the default config, it unregisters from the ring on shutdown so there's no action to take. I would clarify it.

Author: i think i addressed this in https://github.com/grafana/mimir/pull/3698/commits/d8b3b96a738fb51838bfdb205d881349462cc103

   and click "Forget" on the instances that you scaled down.
   Alternatively, you can wait for 10 times the value of `-store-gateway.sharding-ring.heartbeat-timeout` (1m by default)

>    for the store-gateways to be automatically forgotten from the ring.

Reviewer: As I re-read this, I do not like my initial suggestion and recommend the second suggestion.

>    for the store-gateways to be automatically forgotten from the ring.

Author: i think i addressed this in https://github.com/grafana/mimir/pull/3698/commits/d8b3b96a738fb51838bfdb205d881349462cc103

i went with the second suggestion

> 1. Proceed with the next two store-gateway replicas or the next zone when using zone-aware replication.

Reviewer: ```suggestion
1. If you are using zone-aware replication, proceed with the next two store-gateway replicas or the next zone.
```

Reviewer: In line 141 (https://github.com/grafana/mimir/pull/3698/files#diff-b2b4912b253f31f2422ed4f97255308f7b339700558844adf1ea0965a70292f6R141), there is an "if" and here there is a "when". Does "when" mean "if" or "while" here?

Author: ```suggestion
1. Proceed with the next two store-gateway replicas. If you are using zone-aware replication, the proceed with the next zone.
```
